CMSgov/design-system
# [WNMGDS-2690] Use new JSX transform

## Summary

[WNMGDS-2690](https://jira.cms.gov/browse/WNMGDS-2690)

Building on Sarah's work in https://github.com/CMSgov/design-system/pull/2176, which we had to put on hold because of our support for React 16, this change switches to the new JSX transform and removes unnecessary React imports from our JSX modules. This ends up cutting the bundle size by about 4%. 

## How to test

1. Check out 8e27722465115d1263b14f05f62fab540d3cd05b
2. Build and run the `examples/react-app` example, showing that a downstream project can still use the classic JSX transform after we've started using the new one in our library.

## Checklist
